web/app/Models/Interaction.php

```diff
@@ -61,73 +61,148 @@ public function target() : MorphTo
-         */
-        if(is_subclass_of($setter, 'Illuminate\Database\Eloquent\Model')
-            && InteractionType::isValue($relation)
-            && is_subclass_of($target, 'Illuminate\Database\Eloquent\Model'))
+        if(self::isInvalid($setter, $relation, $target, $metadata))
         {
-            $interaction = new Interaction(
-                [
-                    'setter_type' => get_class($setter),
-                    'setter_id' => $setter->id,
-                    'interaction_type' => $relation,
-                    'target_type' => get_class($target),
-                    'target_id' => $target->id,
-                    'metadata' => $metadata
-                ]
-            );
+            Log::error("Interaction::add was given an invalid set of params. 
+            Setter: $setter Relation: $relation Target: $target Metadata: $metadata");
+            abort(500);
+            return false;
+        }
 
-            try {
-                $interaction->save();
-                return $interaction;
-            }
-            catch(QueryException $e) {
-                /** Postgres unique constraint violation: */
-                if($e->errorInfo[0] == 23505) {
-                    /**
-                     * We'll want to throw something here that a controller can
-                     * catch and return data to the user. Pending API work.
-                     */
-                    return null;
-                }
+        $interaction = new Interaction(
+            [
+                'setter_type' => get_class($setter),
+                'setter_id' => $setter->id,
+                'interaction_type' => $relation,
+                'target_type' => get_class($target),
+                'target_id' => $target->id,
+                'metadata' => $metadata
+            ]
+        );
+
+        try {
+            return $interaction->save();
+        }
+        catch(QueryException $e) {
+            /** Postgres unique constraint violation: */
+            if($e->errorInfo[0] == 23505) {
+                /**
+                 * We'll want to throw something here that a controller can
+                 * catch and return data to the user. Pending API work.
+                 */
+                return false;
             }
         }
-        Log::error("Interaction::add was given an invalid set of params. 
+        return false;
+    }
+
+    /**
+     * Retrieve the actual Interaction object rather than the downstream objects.
+     * @param $setter
+     * @param $relation
+     * @param $target
+     * @return Interaction|null
+     */
+    public static function retrieve($setter, $relation, $target) : ?Interaction
+    {
+        if(self::isInvalid($setter, $relation, $target))
+        {
+            Log::error("Interaction::retrieve was given an invalid set of params. 
             Setter: $setter Relation: $relation Target: $target");
-        abort(500);
-        return null;
+            abort(500);
```

Comment:
What does `abort()` do?

Reply:
`abort()` returns a response to the caller with the provided HTTP status code instead of a `die()` statement which does not offer that. This is a stub at the moment, when the controller gets added to the mix we'll throw various exceptions to be provided to the user.